### ISSUE TITLE
wandb error for NERModel()

### ISSUE THREAD

**Issue by nhsmith85** [ThilinaRajapakse/simpletransformers#926]
**Describe the bug**
NERModel() produces error when running wandb sweeps.  

**To Reproduce**
```
!pip install simpletransformers==0.50.0

model_type = 'distilbert'
model_name = 'distilbert-base-uncased'

model_args = NERArgs()

# housekeeping
model_args.reprocess_input_data = True
model_args.evaluate_during_training = True
model_args.evaluate_during_training_steps = -1
model_args.evalute_during_training_verbose = True
model_args.evalute_during_training_silent = False
model_args.evaluate_each_epoch = True
model_args.best_eval_metric = 'f1_score'
model_args.early_stopping_metric = 'f1_score'
model_args.early_stopping_metric_minimize = False
model_args.use_multiprocessing = True
model_args.output_dir = model_dir  + wandb_project
model_args.overwrite_output_dir = True
model_args.best_model_dir = model_dir  + wandb_project + '/best_model'
#model_args.cache_dir = '/tmp/model_cache' 
model_args.manual_seed = 99
model_args.save_steps = -1   # -1 to disable
model_args.save_model_every_epoch = True 
model_args.use_cached_eval_features = False
model_args.wandb_project = wandb_project
model_args.wandb_kwargs = {'name': 'baseline'}
model_args.labels_list = uniq_labels
model_args.classification_report = True

# model specific
model_args.do_lower_case = True    # for uncased set to True
model_args.num_train_epochs = 3
model_args.train_batch_size = 16
model_args.eval_batch_size = 16
model_args.max_seq_length = 50
model_args.learning_rate = 5e-5
model_args.weight_decay = 0.01  # int ?
model_args.warmup_steps = 0

# update args for optimization run
model_args.save_steps = -1   # -1 to disable
model_args.evaluate_during_training = True
model_args.evaluate_each_epoch = True
model_args.save_model_every_epoch = False
model_args.no_save = True
model_args.wandb_kwargs = {'name': 'optimize'}


sweep_config = {
    
    'method': 'grid', #grid, random, bayes
    'metric': {
      'name': 'f1_score',
      'goal': 'maximize'
      },

    'parameters': {
        
        #'learning_rate': {'min': 0, 'max': 5e-5},
        # 'num_train_epochs': {'min': 1, 'max': 10},
        # 'train_batch_size': {'values': [16, 32]}

        'learning_rate': {
            'values': [ 5e-5, 3e-5, 2e-5]
            },
        'train_batch_size': {
            'values': [16, 32]
            },
        'num_train_epochs':{
            'values':[2, 3, 4]
            }
        },

  #  'early_terminate': {'type': 'hyperband', 'min_iter': 2}
}


sweep_id = wandb.sweep(sweep_config, project=wandb_project)


def train():

    # Initialize a new wandb run
    wandb.init()

    # Create a TransformerModel
    model = NERModel(
    model_type = model_type,
    model_name = model_name,
    args = model_args,
    sweep_config = wandb.config
    )

    # Train the model
    model.train_model(train, eval_data = val)
    
    # Sync wandb
    wandb.join()

wandb.agent(sweep_id, train)
```

**Expected behavior**
The train dataset and val dataset run with no errors when I am not running a wandb sweep and just training the model with baseline hyperparam settings. However, when running a **sweep** the error below in the screenshot occurs. Both train and val are dataframes with columns sentence_id, words, labels. 

**Screenshots**
![image](https://user-images.githubusercontent.com/9698634/103324643-63db8980-4a16-11eb-92db-00141a85c000.png)

**Desktop (please complete the following information):**
 - OS

**Additional context**
